### PR TITLE
Database dump and push to S3

### PR DESCRIPTION
**Pull request by nomadicoder**

### CODE REVIEW

File: manifold_database_dump_dag.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# from airflow.operators.python_operator import PythonOperator
+from manifold_airflow_dags.task_slack_posts import slackpostonfail
+# from manifold_airflow_dags.task_slack_posts import slackpostonsuccess
+
+MANIFOLD_INSTANCE_SSH_CONN = \
+        BaseHook.get_connection("AIRFLOW_CONN_MANIFOLD_SSH_INSTANCE")
+MANIFOLD_DUMP_DATABASE_INTERVAL = \
+        Variable.get("MANIFOLD_DUMP_DATABASE_SCHEDULE_INTERVAL")
+DUMP_PATH = "/tmp"
+DUMP_FILENAME = f"manifold-{ datetime.now().strftime('%Y-%m-%dT%H%z')}.sqlc"
```
Comment (bibliotechy) [inline manifold_database_dump_dag.py]: So, each time airflow runs a task in a dag, it loads this dag file and evaluates it. Meaning this `datetime.now()` is going to vary a bit with each task, so the DUMP_FILENAME in the first task may not be the same as the DUMP_FILENAME in the last task. Looks like we are using `execution_date` inside a template elsewere to handle these cases https://github.com/tulibraries/funcake_dags/blob/qa/dev_combine_index_dag.py#L23

File: manifold_database_dump_dag.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+#
+# Data Bucket Variables
+#
+
+AIRFLOW_S3 = BaseHook.get_connection("AIRFLOW_S3")
+AIRFLOW_DATA_BUCKET = Variable.get("AIRFLOW_DATA_BUCKET")
+sftp_put_db_dump_to_s3 = SFTPToS3Operator(
+    task_id='sftp_put_db_dump_to_s3',
+    s3_key=AIRFLOW_S3.login,
```
Comment (bibliotechy) [inline manifold_database_dump_dag.py]: Key here is not the s3 login information. Key is the term s3 uses for it's files. This should be the filename, including directory paths, or "prefixes" as s3 refers to them. 

Shoudl probably be something like f`manifold-db-backups/{DUMP_FILENAME}`

File: variables.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
    "MANIFOLD_HOURS_SYNC_SCHEDULE_INTERVAL": "@daily",
   "MANIFOLD_BLOGS_SYNC_SCHEDULE_INTERVAL": "@daily",
+  "MANIFOLD_DUMP_DATABASE_SCHEDULE_INTERVAL": "@daily",
```
Comment (bibliotechy) [inline variables.json]: Shouldn't this be weekly?

File: manifold_airflow_dags/manifold_database_dump_dag.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    command=dump_database_bash,
+    ssh_conn_id="manifold-db",
+    dag=DAG
+)
+
+#
+# Copy the dump file to the worker so we have a local copy
+#
+
+SFTP_GET_DB_DUMP = SFTPOperator(
```
Comment (bibliotechy) [inline manifold_airflow_dags/manifold_database_dump_dag.py]: Why are you copying the file locally if you are then using SFTPtos3 to transfer the dump to the bucket? We're just going to accumulate db dumps on the worker until we run out of space?

Comment (nomadicoder) [reply manifold_airflow_dags/manifold_database_dump_dag.py]: Removed

**Pull request merged**